File: server/src/main/java/com/example/server/model/Exam.java
package com.example.server.model;

import com.example.server.bll.ExamExpiredResolver;
import com.example.server.model.common.BaseEntity;
import org.babyfish.jimmer.sql.*;

import java.time.LocalDateTime;


@Entity
@Table(name = "exam")
public interface Exam extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id();

    long peopleId();

    @ManyToOne
    People people();

    long paperId();

    @ManyToOne
    Paper paper();

    boolean submitted();

    LocalDateTime startTime();

    @Transient(ExamExpiredResolver.class)
    boolean expired();
}

File: website/src/__generated/model/entities/Exam.ts
import type { Paper, People } from './';

export interface Exam {
    
    readonly createdTime: string;
    
    readonly modifiedTime: string;
    
    readonly id: number;
    
    readonly peopleId: number;
    
    readonly people: People;
    
    readonly paperId: number;
    
    readonly paper: Paper;
    
    readonly submitted: boolean;
    
    readonly marked: boolean;
    
    readonly startTime: string;
    
    readonly expired: boolean;
}
